shared/bls/bls.go

```diff
@@ -125,6 +126,18 @@ func AggregateSignatures(sigs []*Signature) *Signature {
 	return &Signature{val: g1.AggregateSignatures(ss)}
 }
 
+// HashG2WithDomainCompressed returns the compressed hash.
+func HashG2WithDomainCompressed(msg [32]byte, domain uint64) [96]byte {
```

Comment:
Why do we need this?

Reply:
This is for the bls test vectors @prestonvanloon . It requires both the compressed and uncompressed hash